### ISSUE TITLE
Finetune on LIBERO-Spatial only

### ISSUE THREAD

**Issue by Heartache-Doctor** [baaivision/UniVLA#39]
hi, I'm trying to finetune your post training model only on LIBERO-Spatial training data with 8 A100 80G GPU and encounter a few questions.
- I noticed your training log(https://huggingface.co/Yuqi1997/UniVLA/blob/main/UNIVLA_LIBERO_VIDEO_BS192_8K/trainer_state.json) trained 444 epochs in 8000 steps, which is quite different from other VLA models. Could you please tell me the relationship between the number of training epochs and the number of training steps? Is the batch size the same as what you mentioned in paper(192)? I have tried to set my train strategy into epochs and it turns out that 20 epochs can be trained in 40 steps and each steps takes about 1 minute. Since LIBERO-Spatial has about 50000 training samples, it should be about 260 steps = 1 epoch with a batch size of 192. I'm quite confused about what has been ignored by me. Here is my training script:
```
WORLD_SIZE=${WORLD_SIZE:-1}
RANK=${RANK:-0}
MASTER_ADDR=${MASTER_ADDR:-127.0.0.1}
MASTER_PORT=${MASTER_PORT:-23456}
NGPUS=8

DATAPATH='/data/sj/libero_univla/meta/libero_all_norm.pkl'
ACTION_TOKENIZER_PATH="/data/sj/UniVLA/pretrain/fast"
EXP_NAME="UNIVLA_LIBERO_VIDEO_BS192_8k"

export PYTHONPATH=$(pwd)

torchrun \
    --nproc_per_node=${NGPUS} \
    --nnodes=1 \
    --node_rank=${RANK} \
    train/train_moe.py \
    --model_name_or_path /data/sj/UniVLA/pretrain/WORLD_MODEL_POSTTRAIN\
    --model_config_path /data/sj/UniVLA/configs/moe_fast_video.json \
    --deepspeed scripts/sft/zero3_offload.json \
    --output_dir "logs/"${EXP_NAME} \
    --learning_rate 8e-5 \
    --null_prompt_prob 0.15 \
    --weight_decay 0.1 \
    --min_learning_rate 5e-6 \
    --max_grad_norm 5.0 \
    --adam_beta1 0.9 \
    --adam_beta2 0.95 \
    --adam_epsilon 1e-6 \
    --bf16 True \
    --tf32 True \
    --data_path ${DATAPATH} \
    --num_train_epochs 20\
    --dataloader_num_workers 12 \
    --lr_scheduler_type "cosine_with_min_lr" \
    --warmup_steps 50 \
    --per_device_train_batch_size 3 \
    --frames 2 \
    --action_frames 10 \
    --max_position_embeddings 3200 \
    --seed 42 \
    --logging_steps 20 \
    --gradient_checkpointing True \
    --gradient_accumulation_steps 8 \
    --save_strategy epoch \
    --eval_strategy no \
    --apply_loss_on_only_vision False \
    --apply_loss_on_only_action True \
    --actions True \
    --actions_format "fast" \
    --use_gripper True \
    --video_format "interleave" \
    --action_tokenizer_path ${ACTION_TOKENIZER_PATH} \
    --save_total_limit None
```

- I also found that the training hyperparameters for all LIBERO data don't work for LIBERO-Spatial(especially when the training epochs are far small than 444). Could kindly give me a version of training hyperparameters only on LIBERO-Spatial? I paste my training log below, the loss is far bigger than those recorded in your training log:
```
{'loss': 13.0062, 'grad_norm': 20.77370834350586, 'learning_rate': 3.2000000000000005e-05, 'epoch': 8.89}
```

Thanks a lot!

**Comment by Robertwyq** [comment]
@Heartache-Doctor 
1. The approach to data organization differs significantly. Previous methods are image - based, so each frame is split into individual samples for training. In contrast, our method is video - based, where each trajectory constitutes one single sample. This explains why we train for a large number of epochs, especially considering that Libero actually has a relatively small number of training trajectories.
Training the 7B model tends to be quite slow. I have been using 3 to 4 machines with a batch size of 192, and the number of training epochs is not explicitly specified. Additionally, if you only train the spatial suite, the training steps may not require to 8000.

2. I trained them together and haven't tuned the hyperparameters for training a single suite separately. However, the performance should still be able to reach above 95. The issue here is that the loss is too large—could this be due to non-convergence or another reason? If you're only training the spatial component, I suggest you first try training for fewer steps or reducing the batch size. The initial loss might be a bit high, but it depends on how it decreases in the end.

**Comment by xiao10ma** [comment]
I tried to train only on spatial, and its loss decreased as expected.

<img width="1009" height="673" alt="Image" src="https://github.com/user-attachments/assets/371b0acc-9f79-41de-927c-23719af4108b" />

And the hyper parameters followed the full libero dataset.

**Comment by Robertwyq** [comment]
@xiao10ma 
It looks correct; you can check the performance later.
